src/core/reopt.jl

```diff
@@ -355,12 +355,21 @@ function build_reopt!(m::JuMP.AbstractModel, p::REoptInputs)
 		add_to_expression!(Costs, m[:ExpectedOutageCost] + m[:mgTotalTechUpgradeCost] + m[:dvMGStorageUpgradeCost] + m[:ExpectedMGFuelCost])
 	end
 
-	if !p.s.settings.add_soc_incentive
-		@objective(m, Min, m[:Costs])
-	else  # Keep SOC high
-		@objective(m, Min, m[:Costs] - sum(m[:dvStoredEnergy][b, ts] for b in p.s.storage.types.elec, ts in p.time_steps) /
-									   (8760. / p.hours_per_timestep)
+	@objective(m, Min, m[:Costs])
+	
+	if !(isempty(p.s.storage.types.elec)) && p.s.settings.add_soc_incentive # Keep SOC high
+		@objective(m, Min, m[:Costs] - 
+		sum(m[:dvStoredEnergy][b, ts] for b in p.s.storage.types.elec, ts in p.time_steps) /
+			(8760. / p.hours_per_timestep)
```

Comment:
If the battery degrades more with a higher SOC, then we may have a conflicting objective and the incentive in 362-363 may not be enough.  I'll propose we add a warning here that the SOC incentive may need tuning if the instance also includes a degradation model.

Reply:
The objective gets replaced in contraints/battery_degradation with
`@objective(m, Min, m[:Costs] + m[:degr_cost])`
so no SOC incentive is included with degradation.